core/src/test/java/nu/yona/server/analysis/service/ActivityServiceTest.java

```diff
@@ -142,16 +152,22 @@ public void setUp()
 		Set<Goal> goals = new HashSet<>(Arrays.asList(gamblingGoal, gamingGoal, socialGoal, shoppingGoal));
 		deviceAnonEntity = DeviceAnonymized.createInstance(0, OperatingSystem.ANDROID, "Unknown", 0, Optional.empty(),
 				Translator.EN_US_LOCALE);
+
 		userAnonEntity = UserAnonymized.createInstance(anonMessageDestinationEntity, goals);
 		userAnonEntity.addDeviceAnonymized(deviceAnonEntity);
 		UserAnonymizedDto userAnon = UserAnonymizedDto.createInstance(userAnonEntity);
 		userAnonZone = userAnon.getTimeZone();
 		userAnonId = userAnon.getId();
 
 		userId = UUID.randomUUID();
+		UserDto user = UserDto.createInstance(Optional.empty(), "John", "Doe", "+31612345678", "JD", Optional.empty());
+		userCreationTimeField.set(user, Optional.of(LocalDateTime.of(2017, 1, 1, 12, 0)));
```

Comment:
Why not pass those arguments via a `createInstance` method? Reflection is not the nicest language feature.

Reply:
Indeed, but this is a test. It sets up an object in a certain way as a test scenario. For that (and for assertions), we regularly use reflection. I favor that over cluttering the API of the production code. Makes sense?

Reply:
I would prefer a non-public method in that case.

Reply:
And call that through reflection?

Reply:
How about using `@VisibleForTesting`? Reading about it [here](https://medium.com/@vadimpushtaev/how-to-test-private-methods-4bc57d4410ff) now...